cmd/vic-machine/list/list.go

```diff
@@ -178,7 +178,7 @@ func (l *List) Run(clic *cli.Context) (err error) {
 	}
 
 	executor := management.NewDispatcher(validator.Context, validator.Session, management.ListAction, false)
-	vchs, err := executor.SearchVCHs(validator.ClusterPath)
+	vchs, err := executor.SearchVCHs(validator.Session.ClusterPath)
```

Comment:
1. `validator.ClusterPath` never seems to be touched (at least not in this path). `validator.NewValidator` [does not explicitly set it](https://github.com/vmware/vic/blob/c6ea3e26caeefcf0e2df7c842824492e64424885/lib/install/validate/validator.go#L143).

2. It does appear so, as the api calls `NewValidator` as well. I will make a more broad change to include the API and any other users referencing this field.

Reply:
Upon further investigation, I can find no references to `validator.ClusterPath` aside from the list API and from this code here that I've since changed.

We have a couple of options:
1. We could remove the field or fields from the `Validator` struct (and possibly break consistency with the other fields: `TargetPath`, `DatacenterPath`, `ResourcePoolPath`, + 7 more).
2. We could investigate whether or not `validator.Session` is/should be used for this purpose, and if not, set all applicable fields in `Validator` instead. Otherwise, use `Session` - and question whether we need all the fields in `Validator` (see above).
3. Set `ClusterPath` in both `Validator` and `validator.Session` which has a particularly pointless and/or confusing odor to it.

Reply:
For now, since you mentioned more sweeping changes to vic-machine in the 1.5 work, I've opted to simply set the `validator.ClusterPath` in `validate.NewValidator` and leave the `vic-machine ls` CLI & API code as-is.

We can sync up to answer some of these questions on the way to 1.5.

Reply:
I did a little digging.

When `validator.ClusterPath` was introduced (#730), it was used. Shortly after that (#1127 ([diff](https://github.com/vmware/vic/pull/1127/files#diff-0020eaf71e44bdf92a400112eb03adc4L80))), it seems like the code to set it was removed. This would suggest we should just get rid of it.

On the other hand, there's this hint, which suggests an interest in moving _away_ from `Session`-based `*Path` fields in the direction of the ones on `Validator`, which would suggest maybe we should go with #3.:

https://github.com/vmware/vic/blob/c6ea3e26caeefcf0e2df7c842824492e64424885/lib/install/validate/compute.go#L161-L166

I think we may need input from someone who knows the (as far as I can tell) undocumented vision for these structs.